From 74caa878ed8fbbfd743d4db837daee13910951de Mon Sep 17 00:00:00 2001
From: Kurt Lee <kurt.lee@cypress.com>
Date: Tue, 9 Mar 2021 02:18:13 -0600
Subject: [PATCH 072/100] brcmfmac: Fix connecting enterprise AP failure

Use alternative to check AKM-DPP enabled thus this change won't let
status un-sync on use_fwsup between driver and firmware

Signed-off-by: Kurt Lee <kurt.lee@cypress.com>
Signed-off-by: Chung-Hsien Hsu <chung-hsien.hsu@infineon.com>
Signed-off-by: Chi-hsien Lin <chi-hsien.lin@infineon.com>
---
 drivers/net/wireless/broadcom/brcm80211/brcmfmac/cfg80211.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/drivers/net/wireless/broadcom/brcm80211/brcmfmac/cfg80211.c b/drivers/net/wireless/broadcom/brcm80211/brcmfmac/cfg80211.c
index 367d54a0dfd4..b512f7770672 100644
--- a/drivers/net/wireless/broadcom/brcm80211/brcmfmac/cfg80211.c
+++ b/drivers/net/wireless/broadcom/brcm80211/brcmfmac/cfg80211.c
@@ -2023,6 +2023,7 @@ brcmf_set_key_mgmt(struct net_device *ndev, struct cfg80211_connect_params *sme)
 			break;
 		case WLAN_AKM_SUITE_DPP:
 			val = WFA_AUTH_DPP;
+			profile->use_fwsup = BRCMF_PROFILE_FWSUP_NONE;
 			break;
 		default:
 			bphy_err(drvr, "invalid cipher group (%d)\n",
@@ -2350,8 +2351,6 @@ brcmf_cfg80211_connect(struct wiphy *wiphy, struct net_device *ndev,
 				brcmf_dbg(INFO, "using PSK offload\n");
 				profile->use_fwsup = BRCMF_PROFILE_FWSUP_PSK;
 			}
-		} else {
-			profile->use_fwsup = BRCMF_PROFILE_FWSUP_NONE;
 		}
 
 		if (profile->use_fwsup != BRCMF_PROFILE_FWSUP_NONE) {
-- 
2.17.1

